src/SubStatus.h

```diff
@@ -45,21 +48,55 @@
 /// - TransactionSubsMgr::getFullStatus() always returns one of these objects with the std::optional<BlockHeight> as
 ///   the active value, that is, blockHeight() will always be a valid pointer (even if it itself !has_value()).
 ///
+/// - ScriptHashTransactionSubsMgr::getFullStatus() always returns one of these objects with the std::optional<HashSet> as
+///   the active value, that is, hashSet() will always be a valid pointer (even if it itself !has_value()).
+///
+
+using HashSet = std::set<HashX>;
```

Comment:
The new SubStatus 'type' HashSet - it is an ordered set, so that the hashes of two sets are the same for the same transaction composition. Also set comparison will then be trivial.